src/main/java/com/imsweb/x12/Loop.java

```diff
@@ -584,6 +590,43 @@ public String toString() {
         return dump.toString();
     }
 
+    /**
+     * Returns the Loop in X12 String format. This method is used to convert the X12 object into a X12 transaction.
+     * @return String representation
+     */
+    public String toX12String(LoopDefinition loopDefinition) {
+        StringBuilder dump = new StringBuilder();
+
+        Set<Positioned> segmentsAndLoops = new TreeSet<>();
+        if (loopDefinition.getLoop() != null) {
+            segmentsAndLoops.addAll(loopDefinition.getLoop());
+        }
+        if (loopDefinition.getSegment() != null) {
+            segmentsAndLoops.addAll(loopDefinition.getSegment());
+        }
+        for (Positioned positioned : segmentsAndLoops) {
+            if (positioned instanceof SegmentDefinition) {
+                SegmentDefinition segmentDefinition = (SegmentDefinition)positioned;
+                int idx = 0;
+                Segment segment;
+                while ((segment = getSegment(segmentDefinition.getXid(), idx++)) != null) {
+                    dump.append(segment);
+                    dump.append(_separators.getSegment());
+                    dump.append(_separators.getLineBreak().getLineBreakString());
+                }
+            }
+            else if (positioned instanceof LoopDefinition) {
+                LoopDefinition innerLoopDefinition = (LoopDefinition)positioned;
+                int idx = 0;
+                Loop innerLoop;
+                while ((innerLoop = getLoop(innerLoopDefinition.getXid(), idx++)) != null) {
```

Comment:
Using the LoopDefinition like this causes some subtle issues. Using X222 as an example, Loop 2300 can be a child loop of 2000B or 2000C depending on whether the patient is the subscriber or dependent of a subscriber. If the patient is the subscriber then 2000C is not present and Loop 2300 is a child loop of 2000B. If the patient is a dependent of the subscriber then loop 2300 will be a child loop of 2000C but 2000B is still required. Therefore, the LoopDefinition actually contains two definitions of Loop 2300 and all its child loops, one for when Loop 2000C is present and one for when it is not. As a result, if a file contains Loop 2000C I think you will get Loop 2300 and all the child loops printed twice, once after Loop 2010BB and once after Loop 2010CA.  I think we need a check to confirm that the loop we get from getLoop is actually a direct child of the current loop we are processing from the LoopDefinition.

Reply:
@angelaszek ok the x12_complex example seems to reproduce this issue. I will fix it when i get time.

Reply:
![image](https://user-images.githubusercontent.com/2238777/75918244-ff18c800-5e20-11ea-97f1-779fd32c096e.png)

@angelaszek does this failing test jive with what you were thinking?